applications/OptimizationApplication/python_scripts/utilities/opt_convergence.py

```diff
@@ -50,23 +58,26 @@ def __init__(self, parameters: Kratos.Parameters, optimization_problem: Optimiza
         self.__optimization_problem = optimization_problem
         self.__tolerance = parameters["tolerance"].GetDouble()
 
+    @time_decorator()
     def IsConverged(self) -> bool:
         iter = self.__optimization_problem.GetStep()
-        conv = True if iter >= self.__max_iter else False
+        self.conv = True if iter >= self.__max_iter else False
 
         algorithm_buffered_data = ComponentDataView("algorithm", self.__optimization_problem).GetBufferedData()
         if not algorithm_buffered_data.HasValue("search_direction"):
             raise RuntimeError(f"Algorithm data does not contain computed \"search_direction\".\nData:\n{algorithm_buffered_data}" )
 
-        norm = KratosOA.ExpressionUtils.NormL2(algorithm_buffered_data["search_direction"])
-        if not conv:
-            conv = True if norm <= self.__tolerance else False
+        self.norm = KratosOA.ExpressionUtils.NormL2(algorithm_buffered_data["search_direction"])
+        if not self.conv:
+            self.conv = True if self.norm <= self.__tolerance else False
```

Comment:
BTW, isn't this check should be changed? if it reached the `self.__max_iter` at line 64, `self.conv` is true. Then this is not executed at all. So that means are you returning converged `True` if it reaches the `self.__max_iter`?

I would use something like the following where no checks for iterations are done. If the norm is within the range, then the iterations should stop, hence no need to check for iteration.
```python
self.conv = self.norm <= self.__tolerance
```

Reply:
"BTW, isn't this check should be changed? if it reached the self.__max_iter at line 64, self.conv is true. Then this is not executed at all. So that means are you returning converged True if it reaches the self.__max_iter?" -> Yes, so I can stop the opt. if it takes too many iterations.